core/trino-main/src/main/java/io/trino/metadata/AbstractPropertyManager.java

```diff
@@ -69,6 +73,76 @@ protected final void doRemoveProperties(K propertiesKey)
         connectorProperties.remove(propertiesKey);
     }
 
+    protected final Properties doGetProperties(
```

Comment:
If we allow a user to set a property to `DEFAULT` in all the places where a user can set a property, then we only need one version of `doGetProperties`. But we need to decide whether that is indeed what we want first.

Reply:
We need different versions to support the different use cases. For example, we need a version that takes the input properties as a `Map<String, Expression>` (rather than a `List<Property>`) to support `TableProceduresPropertyManager.getProperties`.